README.md

```diff
@@ -21,14 +22,62 @@ The cluster-insight service does not require any changes to the cluster, and sho
-* Run the installation script from the local directory, using a command like `./cluster-insight/install/cluster-insight-setup.sh <path/to/kubernetes>`, where `<path/to/kubernetes>` is the path to your kubernetes binary directory.
+### Easy installation: run the provided installation script
 
-The installation script starts the cluster-insight service and two replication controllers, one for the master and one for the minions. By default, the master replication controller runs one replica, and the minion replication controller runs a replica for every node in the cluster. Both replications controllers pull the latest version of the cluster-insight container image from Docker Hub when necessary.
+The provided installation script, `cluster-insight-setup.sh`, will install and run the service on a cluster.
+You can run it as often as you like to update the service (e.g., when the container image in Docker Hub or any of the configuration files in the
+`./cluster-insight/install` directory have changed).
+The script pulls the latest released version of the image from Docker Hub only if it does not exist yet in the nodes of your cluster.
+To run the script:
+
+* Clone the cluster-insight sources from Github into a local directory named `./cluster-insight`, using a command like
+`git clone https://github.com/google/cluster-insight.git`.
+* Run the installation script from the local directory, using a command like
+`./cluster-insight/install/cluster-insight-setup.sh [<path/to/kubernetes>]`, where the optional `<path/to/kubernetes>` is the path to
+your kubernetes binary directory.
+
+The installation script starts the cluster-insight service and two replication controllers, one for the master and one for the minions.
+By default, the master replication controller runs one replica, and the minion replication controller runs a replica for every node in the cluster.
+Both replications controllers pull the latest released version of the cluster-insight container image from Docker Hub when necessary.
+
+### Behind the scenes: released image tags
+
+The Cluster-Insight images on Docker Hub are tagged by the date they were built. For example, the tag is `2015-07-01` for the image built on
+July 1st, 2015.
+The template files in `./cluster-insight/install/cluster-insight-{master,minion}-controller.yaml` mention the latest released image by its tag.
+We will update the image tag in these template files whenever we push a new released image to Docker Hub.
+
+### For experts only: run the Cluster-Insight in debug mode
+
+The provided installation script, `cluster-insight-setup.sh`, can also install the latest `kubernetes/cluster-insight` image in each node
+(as opposed to the latest released version from Docker Hub), and run then in debug mode.
+To install and run the latest image in each node you should:
```

Comment:
I made it clearer.
Note that you cannot run the installation script on a workstation because the script also starts kubectl in proxy mode, which can work correctly from a workstation only if the firewall rules are set correctly.